plugins/modules/foreman_templates_import.py

```diff
@@ -0,0 +1,185 @@
+    type: str
+  associate:
+    description: Associate to OSes, Locations and Organizations based on metadata.
+    required: false
+    type: str
+    choices:
+     - always
+     - new
+     - never
+  verbose:
+    description: Add template diffs to the output.
+    required: false
+    type: bool
+  force:
+    description: Update templates that are locked.
+    required: false
+    type: bool
+  lock:
+    description: Lock imported templates.
+    required: false
+    type: bool
+  branch:
+    description: Branch in Git repo.
+    required: false
+    type: str
+  repo:
+    description: Filesystem path or repo (with protocol), for example /tmp/dir or git://example.com/repo.git or https://example.com/repo.git.
+    required: false
+    type: str
+  filter:
+    description:
+      - Sync only templates with name matching this regular expression, after I(prefix) was applied.
+      - Case-insensitive, snippets are not filtered.
+    required: false
+    type: str
+  negate:
+    description: Negate the prefix.
+    required: false
+    type: bool
+  dirname:
+    description: The directory within Git repo containing the templates.
+    required: false
+    type: str
+  locations:
+    description: REPLACE locations with given list.
+    required: false
+    type: list
+    elements: str
+  organizations:
+    description: REPLACE organizations with given list.
+    required: false
+    type: list
+    elements: str
+extends_documentation_fragment:
+  - foreman
+'''
+
+EXAMPLES = '''
+- name: Sync templates from git repo
+  foreman_templates_import:
+    repo: https://github.com/theforeman/community-templates.git
+    branch: 1.24-stable
+    associate: new
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "changeme"
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanAnsibleModule
+
+
+def main():
+    module = ForemanAnsibleModule(
+        foreman_spec=dict(
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
+            associate=dict(choices=['always', 'new', 'never']),
+            prefix=dict(),
+            branch=dict(),
+            repo=dict(),
+            filter=dict(),
+            dirname=dict(),
+            verbose=dict(type='bool'),
+            force=dict(type='bool'),
+            lock=dict(type='bool'),
+            negate=dict(type='bool'),
+            locations=dict(type='entity_list', flat_name='location_ids'),
+            organizations=dict(type='entity_list', flat_name='organization_ids'),
+        ),
+        supports_check_mode=False,
+    )
+
+    module_params = module.clean_params()
+    module.connect()
+
+    if 'template' in module.foremanapi.resources or 'templates' in module.foremanapi.resources:
+        resource_name = 'template'
+    else:
+        raise Exception('The server does not seem to have the foreman_templates plugin installed.')
```

Comment:
I think, this should be `fail_json`, or do we catch exceptions if we go with `with module.connect()`?

Reply:
I agree, modules should fail the module way

Reply:
this can be `required_plugins=[('templates', ['*'])]` in `__init__` these days

Reply:
@evgeni, I'd still this if-else in the module because I don't know which resource to use with resource_action. Or is there a smarter way? Maybe move the kludge to the helper (`if resource=="blah" in the resource_action func)  and leave the module clean?

Reply:
and yeah, the if/else is still needed, sure.

or, we could add a api patch (as we do with other plugins) to always rename it to `templates`?

Reply:
You can add something like this to `ForemanAnsibleModule`

```python
_patch_templates_resource_name(self):
    if 'template' in self.foremanapi.apidoc['docs']['resources']:
        self.foremanapi.apidoc['docs']['resources']['templates'] = self.foremanapi.apidoc['docs']['resources']['template']
```

and then call it here:
https://github.com/theforeman/foreman-ansible-modules/blob/36e6e602ff273f76e11f056737c36f98f4b8f059/plugins/module_utils/foreman_helper.py#L390-L391

then no if/else is needed and `require_plugins` will work too.

Reply:
and probably add a comment referring to https://github.com/theforeman/foreman_templates/commit/c1c4756a71fb44e274a7b113dfbc57f77a9549e4 or https://projects.theforeman.org/issues/28750